master/buildbot/reporters/irc.py

```diff
@@ -231,10 +237,10 @@ def groupDescribe(self, channel, action):
 
     def getContact(self, user, channel=None):
         # nicknames and channel names are case insensitive
```

Comment:
Is this the case also for telegram? could this be an attack to be able to fool the bot for greater rights?

Reply:
`getContact` in telegram accepts dict and retrieves numeric id for user and chat (both cannot be faked as they are set inernally by telegram). So this is non-issue here, as one gives numbers to `authz` for telegram. You need either to give our bot the command `/getid` or talk to https://t.me/myidbot to get the user id number. This is even more secure than in case or IRC, as one cannot get the numeric id of another user easily.

The only important thing is to keep the webhook resource path secret, as this is the only way of ensuring that the updates really come from telegram. I believe that Twisted does not provide any way to reveal the resource paths (I would be stupid if it did, but I am not a Twisted expert).

Reply:
ah I though this was in the command code. I indeed saw the /getid command.

Resource path as a secret is okay as long as there is https frontend.